server/aesdsocket.c

```diff
@@ -0,0 +1,253 @@
+/**
+ * @name: aesdsocket.c
+ * @brief: A socket program for a server in stream mode.
+ * @author: Saurav Negi
+ */
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <signal.h>
+#include <syslog.h>
+#include <sys/socket.h>
+#include <netinet/in.h>
+#include <arpa/inet.h>
+#include <fcntl.h>
+#include <sys/types.h>
+
+#define PORT 9000
+#define DATA_FILE "/var/tmp/aesdsocketdata"
+#define BUFFER_SIZE 1024
+#define ERROR_MEMORY_ALLOC   -6
+#define FILE_PERMISSIONS 0644
+
+int server_socket;
+int client_socket;
+int daemon_mode = 0;
+int option_value = 1;
+
+// Function to handle signals
+static void signal_handler(int signo);
+
+// Function to daemonize the process
+void daemonize();
+
+int main(int argc, char **argv);
+
+/**
+ * @function: signal_handler
+ * @brief: Handles signals like SIGINT and SIGTERM.
+ * @params:
+ *    - signo: The signal number.
+ * @return: None
+ */
+static void signal_handler(int signo) {
+    if (signo == SIGINT || signo == SIGTERM) {
+        syslog(LOG_INFO, "Caught signal, exiting"); // Log that a signal was caught
+        close(server_socket); // Close the server socket
+        close(client_socket); // Close the client socket
+        closelog(); // Close syslog
+        remove(DATA_FILE); // Remove the data file
+        exit(0); // Exit the program
```

Comment:
@sane5805 are all of these safe to do in signal handler?  Consider a design where you can handle cleanup outside the signal hander instead.